tests/pkio_test.py

```diff
@@ -59,6 +59,16 @@ def test_has_file_extension():
     pkeq(False, pkio.has_file_extension("filename_with_no_extension", "json"))
 
 
+def test_is_binary():
+    from pykern import pkunit
+    from pykern import pkio
+
+    d = pkunit.data_dir()
+    pkunit.pkeq(False, pkio.is_binary(d.join("empty.dat")))
+    pkunit.pkeq(False, pkio.is_binary(d.join("text.dat")))
+    pkunit.pkeq(True, pkio.is_binary(d.join("binary.dat")))
```

Comment:
The advantage of doing an sans-i/o implementation would be to not have this type of code. You could simply have byte strings. The binary.dat is too complicated. It will always pass muster. You want to find something that's at the edge of the threshold to test.